python/sbp/client/handler.py

```diff
@@ -316,54 +232,37 @@ def wait_callback(self, callback, msg_type=None, timeout=1.0):
       Waiting period
     """
     event = threading.Event()
-    def cb(sbp_msg):
-      callback(sbp_msg)
+    def cb(msg, **metadata):
+      callback(msg, **metadata)
       event.set()
     self.add_callback(cb, msg_type)
     event.wait(timeout)
     self.remove_callback(cb, msg_type)
 
-class LoggerDriver(BaseDriver):
-  """LoggerDriver
-
-  The :class:`LoggerDriver` class connects a logger (typically a
-  TCP/UDP "logger") and a driver (typically a serial port driver).
-
-  """
-  def __init__(self, logger, driver):
-    print "startup here logger driver here thing"
-    self.driver = driver
-    self.logger = self.handle = logger
-    super(LoggerDriver, self).__init__(self.handle)
-
-  def read(self, size):
+  class _SBPQueueIterator(object):
     """
-    Read wrapper.
-
-    Parameters
-    ----------
-    size : int
-      Number of bytes to read.
+    Class for upstream iterators.  Implements callable interface for adding
+    messages into the queue, and iterable interface for getting them out.
     """
-    data = self.driver.read(size)
-    if data is not None:
-      self.logger.write(data)
-    self.write()
-    return data
+    def __init__(self, maxsize):
```

Comment:
This class is not intended to be public.  The default value of an unbounded queue is in Handler.filter()